In [12]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load in 

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the "../input/" directory.
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# Any results you write to the current directory are saved as output.

/kaggle/input/pulsar_stars.csv


Getting the data from the CSV File and splitting into input and predictions.

In [13]:
df = pd.read_csv("/kaggle/input/pulsar_stars.csv")
print(df.head())
values = df.values
print(values.shape)
X = values[:, 0:7]
Y = values[:, 8]

    Mean of the integrated profile  \
0                       140.562500   
1                       102.507812   
2                       103.015625   
3                       136.750000   
4                        88.726562   

    Standard deviation of the integrated profile  \
0                                      55.683782   
1                                      58.882430   
2                                      39.341649   
3                                      57.178449   
4                                      40.672225   

    Excess kurtosis of the integrated profile  \
0                                   -0.234571   
1                                    0.465318   
2                                    0.323328   
3                                   -0.068415   
4                                    0.600866   

    Skewness of the integrated profile   Mean of the DM-SNR curve  \
0                            -0.699648                   3.199833   
1                        

Importing Scikit-learn libraries.
Using SVC as it is a classification problem.
Using accuracy as measurement index.

In [14]:
from sklearn import preprocessing
from sklearn.model_selection import train_test_split
from sklearn.svm import SVC
from sklearn.metrics import accuracy_score
from sklearn.preprocessing import StandardScaler

Splitting into train and test sets.
Also applying a standard scaler.

In [15]:
X_train, X_test, y_train, y_test = train_test_split(X, Y, random_state=33)
scaler = StandardScaler().fit(X_train)
s_X = scaler.transform(X_train)
s_X_test = scaler.transform(X_test)

Creating a Support Vector Classifier with radial basic function kernal.
Then fitting the training dataset.

In [16]:
svc = SVC(kernel='rbf')
svc.fit(s_X, y_train)

SVC(C=1.0, cache_size=200, class_weight=None, coef0=0.0,
    decision_function_shape='ovr', degree=3, gamma='auto_deprecated',
    kernel='rbf', max_iter=-1, probability=False, random_state=None,
    shrinking=True, tol=0.001, verbose=False)

In [17]:
y_pred = svc.predict(s_X_test)
accuracy_score(y_test, y_pred)

0.9792178770949721

We have created a Model which if given the values of the parameters can predict with 97.9218% accuracy if the star is a Pulsar Star.